### ISSUE TITLE
EpollSocketTcpMd5Test.testKeyMismatch failing consistently

### ISSUE THREAD

**Issue by serioussam** [netty/netty#4264]
Does anyone knows why this test would fail? If there is no quick fix I can dig into it. I am running this on 4.0 branch

Machine details
```
[INFO] ------------------------------------------------------------------------
[INFO] Detecting the operating system and CPU architecture
[INFO] ------------------------------------------------------------------------
[INFO] os.detected.name: linux
[INFO] os.detected.arch: x86_64
[INFO] os.detected.release: ubuntu
[INFO] os.detected.release.version: 14.04
[INFO] os.detected.release.like.ubuntu: true
[INFO] os.detected.release.like.debian: true
[INFO] os.detected.classifier: linux-x86_64
```

Failure trace
```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running io.netty.channel.epoll.EpollSocketTcpMd5Test
12:36:32.368 [main] DEBUG i.n.u.i.l.InternalLoggerFactory - Using SLF4J as the default logging framework
12:36:32.371 [main] DEBUG i.n.c.MultithreadEventLoopGroup - -Dio.netty.eventLoopThreads: 16
12:36:32.386 [main] DEBUG i.n.util.internal.PlatformDependent0 - java.nio.Buffer.address: available
12:36:32.387 [main] DEBUG i.n.util.internal.PlatformDependent0 - sun.misc.Unsafe.theUnsafe: available
12:36:32.387 [main] DEBUG i.n.util.internal.PlatformDependent0 - sun.misc.Unsafe.copyMemory: available
12:36:32.387 [main] DEBUG i.n.util.internal.PlatformDependent0 - java.nio.Bits.unaligned: true
12:36:32.388 [main] DEBUG i.n.util.internal.PlatformDependent - Java version: 8
12:36:32.388 [main] DEBUG i.n.util.internal.PlatformDependent - -Dio.netty.noUnsafe: false
12:36:32.388 [main] DEBUG i.n.util.internal.PlatformDependent - sun.misc.Unsafe: available
12:36:32.388 [main] DEBUG i.n.util.internal.PlatformDependent - -Dio.netty.noJavassist: false
12:36:32.490 [main] DEBUG i.n.util.internal.PlatformDependent - Javassist: available
12:36:32.490 [main] DEBUG i.n.util.internal.PlatformDependent - -Dio.netty.tmpdir: /tmp (java.io.tmpdir)
12:36:32.490 [main] DEBUG i.n.util.internal.PlatformDependent - -Dio.netty.bitMode: 64 (sun.arch.data.model)
12:36:32.490 [main] DEBUG i.n.util.internal.PlatformDependent - -Dio.netty.noPreferDirect: false
12:36:32.507 [main] DEBUG i.n.u.internal.NativeLibraryLoader - -Dio.netty.tmpdir: /tmp (java.io.tmpdir)
12:36:32.507 [main] DEBUG i.n.u.internal.NativeLibraryLoader - -Dio.netty.native.workdir: /tmp (io.netty.tmpdir)
12:36:32.518 [main] DEBUG io.netty.util.NetUtil - Loopback interface: lo (lo, 0:0:0:0:0:0:0:1%lo)
12:36:32.519 [main] DEBUG io.netty.util.NetUtil - /proc/sys/net/core/somaxconn: 128
12:36:32.545 [main] DEBUG i.n.util.internal.ThreadLocalRandom - -Dio.netty.initialSeedUniquifier: 0x3225d3e6fe3158c9 (took 6 ms)
12:36:32.575 [main] DEBUG io.netty.buffer.ByteBufUtil - -Dio.netty.allocator.type: unpooled
12:36:32.576 [main] DEBUG io.netty.buffer.ByteBufUtil - -Dio.netty.threadLocalDirectBufferSize: 65536
12:36:32.592 [epollEventLoopGroup-2-1] DEBUG i.n.channel.DefaultChannelPipeline - Discarded inbound message [id: 0xd5677dad, /127.0.0.1:52821 => /127.0.0.1:60498] that reached at the tail of the pipeline. Please check your pipeline configuration.
Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.372 sec <<< FAILURE! - in io.netty.channel.epoll.EpollSocketTcpMd5Test
testKeyMismatch(io.netty.channel.epoll.EpollSocketTcpMd5Test)  Time elapsed: 0.068 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: io.netty.channel.ConnectTimeoutException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:32)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:254)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:149)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)


Results :

Failed tests: 
  EpollSocketTcpMd5Test.testKeyMismatch Expected exception: io.netty.channel.ConnectTimeoutException
```

**Comment by normanmaurer** [comment]
@serioussam strange enough it not fails here :/

**Comment by serioussam** [comment]
Maybe because I am running on Java 8? I will give it a try with java 7 and see if it works. Else I will dig in and see why it fails just on my box.

**Comment by serioussam** [comment]
java version is not an issue. I will investigate whats happening.

**Comment by serioussam** [comment]
Found it and added PR with fix.

**Comment by normanmaurer** [comment]
@serioussam thanks again... Fixed by  #4267
